src/FedoraApi.php

```diff
@@ -305,4 +307,79 @@ public function getGraph(ResponseInterface $response)
         }
         return $graph;
     }
+
+    /**
+     * Creates version in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param string $timestamp Timestamp for Memento version
+     * @param string $content String or binary content
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function createVersion(
+        $uri = '',
+        $timestamp = '',
+        $content = null,
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false];
+        if ($timestamp != ''){
+            $headers['Memento-Datetime'] = $timestamp;
+        }
+        if ($content != null){
+            $options['body'] = $content;
+        }
+        $options['headers'] = $headers;
+
+        return $this->client->request(
+            'POST',
+            $timemap_uri,
+            $options
+        );
+    }
+
+    /**
+     * Gets list of versions in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function getVersions(
+        $uri = '',
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false, 'headers' => $headers];
+
+        return $this->client->request(
+            'GET',
+            $timemap_uri,
+            $options
+        );
+    }
+
+    /**
+     * Helper method to get the Headers for a resource
+     * and parse the timemap header from it
+     * @param string $uri Fedora Resource URI
+     * @param array $header HTTP Headers
+     *
+     * @return string
+     */
+    private function getTimemapURI($uri, $headers){
+        $resource_headers = $this->getResourceHeaders($uri, $headers);
+        $parsed_link_headers = Psr7\parse_header($resource_headers->getHeader('Link'));
+        $timemap_uri = NULL;
+        foreach($parsed_link_headers as $link_header){
+            if (isset($link_header['rel']) && $link_header['rel'] == "timemap") {
+                $timemap_uri = $link_header[0];
+                $timemap_uri = str_replace("<","",$timemap_uri);
+                $timemap_uri = str_replace(">","",$timemap_uri);
```

Comment:
Super minor suggestion, you could replace these two lines with 
```
$timemap_uri = trim($timemap_uri, "<>");
```
You could even include the original whitespace list to ensure any whitespace is removed.
```
$timemap_uri = trim($timemap_uri, "<> \t\n\r\0\x0B");
```

Reply:
cool, thanks

Reply:
looks good: b0661d4